libraries/rush-lib/src/api/RushProjectConfiguration.ts

```diff
@@ -42,6 +42,47 @@ export interface IRushProjectJson {
   operationSettings?: IOperationSettings[];
 }
 
+export interface IRushPhaseSharding {
+  /**
+   * The number of shards to create.
+   */
+  count: number;
+
+  /**
+   * The format of the argument to pass to the command to indicate the shard index and count.
+   * The default value is `--shard={shardIndex}/{shardCount}`.
+   */
+  shardArgumentFormat?: string;
+
+  /**
+   * An optional argument to pass to the command to indicate the output folder for the shard.
+   * The default argument is `--shard-output-folder=.rush/operations/{phaseName}/shards/{shardIndex}`.
+   */
+  outputFolderArgument?: {
+    /**
+     * The format of the parent folder name to output shards to. Child folders will be created for each
+     *  shard index, ie `{parentFolder}/1`, `{parentFolder}/2`. The default value is `.rush/operations/{phaseName}/shards`.
+     */
+    parentFolderName?: string;
+
+    /**
+     * The name of the argument to pass to the command to indicate the output folder for the shard.
+     * The default value is `--shard-output-folder`. Does not accept templating.
+     */
+    parameterLongName: string;
+  };
```

Comment:
For consistency with `shardArgumentFormat`, why not: 

```suggestion
  /**
   * An optional argument to pass to the command to indicate the output folder for the shard.
   *
   * @defaultValue `--shard-output-folder=.rush/operations/{phaseName}/shards/{shardIndex}`
   */
  outputFolderArgumentFormat?: string;
```

Reply:
Call out that this path is relative to the project root.

Reply:
I think we want the argument name to be configurable so you can pass it easily into existing commands and I couldn't think of a good way to make that happen without separating the parameter name and the value.